fix(editor): execute `oxc.path.server` in win32 with shell

Sysix opened the pull request:
closes #14167
This allows cross-platform configuration in a project.
The project can now define `oxc.path.server` to `node_modules/.bin/oxc_language_server` and windows will pick it up too.

 With `validateSafeBinaryPath` in `Pathvalidator.ts`, we should be sure that bad actors can not harm the user's machine.

This is needed because the VSCode extension will not ship the language server in the future.

nrayburn-tech commented on editors/vscode/client/extension.ts:
```diff
@@ -155,10 +154,20 @@ export async function activate(context: ExtensionContext) {
+      // The npm bin script is a shell script that points to the actual binary.
+      // Security: We validated the userDefinedBinary in `configService.getUserServerBinPath()`.
+      shell: process.platform === 'win32' && command === userDefinedBinary &&
+        userDefinedBinary?.endsWith('node_modules\\.bin\\oxc_language_server'),
```
I don’t personally think this (or the other path validations) are necessary.

If the user has opted into the trusted workspace, they have consented to these settings. I don’t think the extension is responsible for validating what the path might look like or what the contents of the evaluated file might be.

It has potential to make it harder for users to use, for what I would consider little to no real security benefit.

camc314 commented:
### Merge activity

* **Nov 14, 1:18 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 14, 1:18 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
* **Nov 14, 1:22 PM UTC**: The [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) removed this pull request due to **downstack failures on PR #[15680](https://app.graphite.com/github/pr/oxc-project/oxc/15680)**.
* **Nov 14, 1:22 PM UTC**: The [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) removed this pull request due to **downstack failures on PR #[15680](https://app.graphite.com/github/pr/oxc-project/oxc/15680)**.

Sysix committed:
fix(editor): execute `oxc.path.server` in win32 with shell (#14203)

closes #14167
This allows cross-platform configuration in a project.
The project can now define `oxc.path.server` to `node_modules/.bin/oxc_language_server` and windows will pick it up too.

 With `validateSafeBinaryPath` in `Pathvalidator.ts`, we should be sure that bad actors can not harm the user's machine.

This is needed because the VSCode extension will not ship the language server in the future.
